src/algorithms/meta/FilterByAssociations.h

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Simon Gardner
+
+#pragma once
+
+#include <spdlog/spdlog.h>
+#include <algorithms/algorithm.h>
+#include <string>
+#include <string_view>
+
+#include "services/log/Log_service.h"
+#include "algorithms/interfaces/WithPodConfig.h"
+
+namespace eicrecon {
+
+   template<class T, class TA>
+     using FilterByAssociationsAlgorithm =  algorithms::Algorithm<
+       typename algorithms::Input<
+        typename T::collection_type,
+        typename TA::collection_type
+        >,
+       typename algorithms::Output<
+        typename TA::collection_type,
+        typename TA::collection_type
+        >
+     >;
+
+
+  template<typename T,typename TA, auto MemberFunctionPtr>
+  class FilterByAssociations : public FilterByAssociationsAlgorithm<T,TA> {
+
+    public:
+    FilterByAssociations(std::string_view name)
+      : FilterByAssociationsAlgorithm<T,TA>{name,
+                        {"inputCollection","associatedCollection"},
+                        {"isAssociatedCollection","isNotAssociatedCollection"},
```

Comment:
I don't think these values matter that much, but for the sake of documentation:
```suggestion
                        {"inputCollection", "inputAssociationsCollection"},
                        {"outputMatchedAssociations", "outputUnmatchedAssociations"},
```